src/pages/filter/common/filter-control/filter-control.tsx

```diff
@@ -48,61 +51,58 @@ export const FilterControl = observer((): ReactElement => {
   }
 
   return (
-    <div className="flex flex-wrap justify-end bg-blue-dark pr-6 pb-4 pl-6">
-      <div className="flex items-center w-full mt-5">
-        {page === GlbPagesNames.Refiner ? (
-          <FilterControlRefiner />
-        ) : (
-          <FilterControlDtree />
-        )}
-      </div>
-
-      <div className="flex justify-between w-full mt-3">
+    <div className="flex flex-wrap justify-end bg-blue-dark pr-4 pb-4 pl-4">
+      <div className="flex justify-between w-full mt-1.5">
         <div className="flex items-center">
-          <span className="text-grey-blue text-14 font-bold mr-2">
-            {t('filter.method')}
-          </span>
-
-          <DropDown
-            options={FilterControlOptions}
-            value={pageName}
-            onSelect={args => {
-              goToPage(args.value as FilterControlOptions)
-            }}
-          />
+          <SolutionDropDown pageName={pageName} goToPage={goToPage} />
 
-          {page === GlbPagesNames.Dtree && (
-            <Button
-              text="Text editor"
-              className="ml-2"
-              variant={'secondary-dark'}
-              onClick={() => modalsVisibilityStore.openModalTextEditor()}
-              dataTestId={DecisionTreesMenuDataCy.textEditor}
-            />
+          <ControlPanelDivider />
+
+          {page === GlbPagesNames.Refiner ? (
+            <FilterControlRefiner />
+          ) : (
+            <FilterControlDtree />
           )}
-          {/* Temporarily removed in Refiner page */}
+
+          <ControlPanelDivider />
+
+          <DatasetCreationButton />
+
           {page === GlbPagesNames.Dtree && (
             <>
+              <ControlPanelDivider />
+
               <Button
-                text="Undo"
-                className="ml-2"
-                variant={'secondary-dark'}
-                disabled={isUndoLocked}
-                onClick={() => moveActionHistory(-1)}
-              />
-              <Button
-                text="Redo"
-                className="ml-2"
-                variant={'secondary-dark'}
-                disabled={isRedoLocked}
-                onClick={() => moveActionHistory(1)}
+                text={t('dtree.textEditor')}
+                size="md"
+                variant="secondary-dark"
+                onClick={() => modalsVisibilityStore.openModalTextEditor()}
+                dataTestId={DecisionTreesMenuDataCy.textEditor}
               />
             </>
           )}
         </div>
-        <Link className="flex flex-wrap" to={Routes.Root}>
-          <Icon name="Close" className="text-white cursor-pointer" />
-        </Link>
+
+        <div className="flex items-center">
+          {/* TODO: implement undo/redo for FR+WS */}
+
+          {page === GlbPagesNames.Dtree ? (
+            <UndoRedoButtons
```

Comment:
Icons seem to be not properly centered (the image below is from Figma). You can use PerfectPixel extension for Chome if it helps
<img width="590" alt="Screen Shot 2022-05-25 at 00 47 45" src="https://user-images.githubusercontent.com/96476703/170137966-7cb9f0f4-417f-4d82-b2ba-a9554aafa487.png">